src/server/pfs/s3/driver.go

```diff
@@ -115,64 +89,31 @@ func bucketNameToCommit(bucketName string) *pfs.Commit {
 		id = parts[0]
 		parts = parts[1:]
 	}
-	if len(parts) > 1 {
-		branch = parts[0]
-		parts = parts[1:]
-	}
-	if len(parts) == 1 {
-		repo = client.NewProjectRepo(pfs.DefaultProjectName, parts[0])
-	} else {
-		repo = client.NewSystemProjectRepo(pfs.DefaultProjectName, parts[1], parts[0])
-	}
-
-	return repo.NewCommit(branch, id)
-}
 
-func bucketNameToProjectCommit(bucketName string) (*pfs.Commit, error) {
-	var id string
-	branch := "master"
-	var repo *pfs.Repo
-
-	// the name is [commitID.][branch. | branch.type.]repoName.projectName
-	// in particular, to access a non-user system repo, the branch name must be given
-	parts := strings.SplitN(bucketName, ".", 5)
-	if uuid.IsUUIDWithoutDashes(parts[0]) {
-		id = parts[0]
-		parts = parts[1:]
-	}
-	if len(parts) > 2 {
-		branch = parts[0]
-		parts = parts[1:]
-	}
 	switch len(parts) {
 	case 0:
-		return nil, errors.Errorf("bad bucket name %s", bucketName)
+		return nil, errors.Errorf("invalid bucket name %q; must include a repo", bucketName)
 	case 1:
-		return nil, errors.Errorf("bad bucket name %s", bucketName)
+		// e.g. s3://myrepo
+		repo, branch = client.NewProjectRepo(pfs.DefaultProjectName, parts[0]), "master"
 	case 2:
-		repo = client.NewProjectRepo(parts[1], parts[0])
+		// e.g. s3://mybranch.myrepo
+		repo, branch = client.NewProjectRepo(pfs.DefaultProjectName, parts[1]), parts[0]
 	case 3:
-		repo = client.NewSystemProjectRepo(parts[2], parts[1], parts[0])
-	default:
```

Comment:
Should the default case have been removed?

Reply:
Nope, thanks! I fixed it and added a test (and, actually, changed the `SplitN(bucketName, 4)` call above to `SplitN(bucketName, 5)` to fix the test—bucket names could have 4 components, so splitting the name into only 4 pieces could mask a too-long-bucket-name error by parsing the name so that the project has a '.' in it. But they can't have 5)